apps/server/src/modules/tldraw/repo/tldraw-board.repo.spec.ts

```diff
@@ -0,0 +1,221 @@
+				const storeGetYDocSpy = jest
+					.spyOn(repo.mdb, 'getYDoc')
+					.mockImplementation(() => Promise.resolve(new WsSharedDocDo('TEST', service)));
+				const byteArray = new TextEncoder().encode(clientMessageMock);
+
+				return {
+					doc,
+					byteArray,
+					storeUpdateSpy,
+					storeGetYDocSpy,
+				};
+			};
+
+			it('should update db with diff', async () => {
+				const { doc, byteArray, storeUpdateSpy, storeGetYDocSpy } = await setup();
+
+				await repo.updateDocument('TEST', doc);
+				doc.emit('update', [byteArray, undefined, doc]);
+				expect(storeUpdateSpy).toHaveBeenCalled();
+				expect(storeUpdateSpy).toHaveBeenCalledTimes(1);
+				storeUpdateSpy.mockRestore();
+				storeGetYDocSpy.mockRestore();
+				ws.close();
+			});
+		});
+	});
+
+	describe('getYDocFromMdb', () => {
+		describe('when taking doc data from db', () => {
+			const setup = () => {
+				const storeGetYDocSpy = jest
+					.spyOn(repo.mdb, 'getYDoc')
+					.mockImplementation(() => Promise.resolve(new WsSharedDocDo('TEST', service)));
+
+				return {
+					storeGetYDocSpy,
+				};
+			};
+
+			it('should return ydoc', async () => {
+				const { storeGetYDocSpy } = setup();
+				expect(await repo.getYDocFromMdb('test')).toBeInstanceOf(Doc);
+
+				storeGetYDocSpy.mockRestore();
+			});
+		});
+	});
+
+	describe('updateStoredDocWithDiff', () => {
+		describe('when the difference between update and current drawing is more than 0', () => {
+			const setup = () => {
+				const calculateDiffSpy = jest.spyOn(YjsUtils, 'calculateDiff').mockImplementationOnce(() => 1);
+				const storeUpdateSpy = jest.spyOn(repo.mdb, 'storeUpdate').mockImplementation(() => Promise.resolve(1));
+
+				return {
+					calculateDiffSpy,
+					storeUpdateSpy,
+				};
+			};
+
+			it('should call store update method', () => {
+				const { storeUpdateSpy, calculateDiffSpy } = setup();
+				const diffArray = new Uint8Array();
+				repo.updateStoredDocWithDiff('test', diffArray);
+
+				expect(storeUpdateSpy).toHaveBeenCalled();
+
+				calculateDiffSpy.mockRestore();
+				storeUpdateSpy.mockRestore();
+			});
+		});
+
+		describe('when the difference between update and current drawing is 0', () => {
+			const setup = () => {
+				const calculateDiffSpy = jest.spyOn(YjsUtils, 'calculateDiff').mockImplementationOnce(() => 0);
+				const storeUpdateSpy = jest.spyOn(repo.mdb, 'storeUpdate');
+
+				return {
+					calculateDiffSpy,
+					storeUpdateSpy,
+				};
+			};
+
+			it('should not call store update method', () => {
+				const { storeUpdateSpy, calculateDiffSpy } = setup();
+				const diffArray = new Uint8Array();
+				repo.updateStoredDocWithDiff('test', diffArray);
+
+				expect(storeUpdateSpy).not.toHaveBeenCalled();
+
+				calculateDiffSpy.mockRestore();
+				storeUpdateSpy.mockRestore();
+			});
+		});
+	});
+
+	describe('flushDocument', () => {
+		const setup = () => {
+			const flushDocumentSpy = jest.spyOn(repo.mdb, 'flushDocument').mockImplementation(() => Promise.resolve());
```

Comment:
.mockResolveValueOnce();  
The mockImplementation for syncron operations. For async use always mockResolveValueOnce or mockRejectValueOnce. Please also use every the the Once if it is executed 2 times in same test, you can simple add multiple mockResolveValueOnce in chain.